divide_and_conquer/closest_pair_of_points.py

```diff
@@ -1,66 +1,113 @@
 """
-The algorithm finds distance between closest pair of points in the given n points.
-Approach: Divide and conquer 
-The points are sorted based on x-cords 
+The algorithm finds distance btw closest pair of points in the given n points.
+Approach used -> Divide and conquer 
+The points are sorted based on Xco-ords 
 & by applying divide and conquer approach, 
 minimum distance is obtained recursively.
 
-Edge case: closest points lie on different sides of partition
+>> closest points lie on different sides of partition
 This case handled by forming a strip of points 
-which are at a distance (< closest_pair_dis) from mid-point.
-(It is a proven that strip contains at most 6 points)
-And brute force method is applied on strip to find closest points. 
+whose Xco-ords distance is less than closest_pair_dis
+from mid-point's Xco-ords.
+Closest pair distance is found in the strip of points. (closest_in_strip)
 
-Time complexity: O(n * (logn) ^ 2)
+min(closest_pair_dis, closest_in_strip) would be the final answer.
+ 
+Time complexity: O(n * (logn)^2)
```

Comment:
If you want you can put a reference to it. Are there any references in CLRS book?